.github/actions/changeset-determine-step/main.ts

```diff
@@ -0,0 +1,43 @@
+import { execSync } from 'child_process';
+
+import fs from 'fs';
+
+import core from '@actions/core';
+
+try {
+    // Collect changesets statss
+    const random = Math.random().toString(36).slice(2, 10);
+    const statusPath = `changeset-status-${random}.json`
+    execSync(`npx changeset status --output "${statusPath}"`, { stdio: 'inherit' });
```

Comment:
nit: we can directly capture stdout and parse. This will make things a bit more complex, but it will be a more accurate and faster approach that doesn't require a temp file.